packages/medusa/src/services/order-edit.ts

```diff
@@ -315,6 +322,130 @@ export default class OrderEditService extends TransactionBaseService {
+   * @param orderEditId
+   * @param originalLineItemId
+   * @param data
+   */
+  async updateLineItem(
+    orderEditId: string,
+    originalLineItemId: string,
+    data: { quantity: number }
+  ): Promise<void> {
+    return await this.atomicPhase_(async (manager) => {
+      const orderEdit = await this.retrieve(orderEditId, {
+        select: [
+          "id",
+          "items",
+          "order_id",
+          "created_at",
+          "requested_at",
+          "confirmed_at",
+          "declined_at",
+          "canceled_at",
+        ],
+        relations: ["items"],
+      })
+
+      const isOrderEditActive = OrderEditService.isOrderEditActive(orderEdit)
+      if (!isOrderEditActive) {
+        throw new MedusaError(
+          MedusaError.Types.NOT_ALLOWED,
+          `Can not update an item on the order edit ${orderEditId} with the status ${orderEdit.status}`
+        )
+      }
+
+      const clonedLineItem = orderEdit.items?.find(
+        (item) => item.original_item_id === originalLineItemId
+      )
+
+      if (!clonedLineItem) {
+        throw new MedusaError(
+          MedusaError.Types.INVALID_DATA,
+          `Invalid line item id ${originalLineItemId} unable to find the line item on the order edit for the order ${orderEdit.order_id}.`
+        )
+      }
+
+      const orderEditItemChangeServiceTx =
+        this.orderEditItemChangeService_.withTransaction(manager)
+
+      let change = (
+        await orderEditItemChangeServiceTx.list(
+          {
+            original_line_item_id: originalLineItemId,
+            type: OrderEditItemChangeType.ITEM_UPDATE,
+          },
+          {
+            select: ["line_item_id", "original_line_item_id"],
+          }
+        )
+      ).pop()
+
+      if (!change) {
+        change = await orderEditItemChangeServiceTx.create({
+          type: OrderEditItemChangeType.ITEM_UPDATE,
+          order_edit_id: orderEditId,
+          original_line_item_id: originalLineItemId,
+          line_item_id: clonedLineItem.id,
+        })
+      }
+
+      await this.lineItemService_
+        .withTransaction(manager)
+        .update(change.line_item_id!, {
+          quantity: data.quantity,
+        })
+
+      await this.refreshAdjustments(orderEditId)
+    })
+  }
+
+  async refreshAdjustments(orderEditId: string) {
+    const manager = this.transactionManager_ ?? this.manager_
+
+    const lineItemAdjustmentServiceTx =
+      this.lineItemAdjustmentService_.withTransaction(manager)
+
+    const orderEdit = await this.retrieve(orderEditId, {
+      relations: [
+        "items",
+        "items.adjustments",
+        "items.tax_lines",
+        "order",
+        "order.customer",
+        "order.discounts",
+        "order.discounts.rule",
+        "order.gift_cards",
+        "order.region",
+        "order.shipping_address",
+        "order.shipping_methods",
```

Comment:
thought(non-blocking): I don't love how we are joining everything from the order here - we should probably change that in the future

Reply:
Me neither 🤣 do you have something in mind? In the future if everything is editable the order edit will probably become an order clone instead 🤔. Cause each time we will allow new properties or relation to be editable they will probably be cloned and then attached to the order edit, so in the end it becomes an order no?

Reply:
yeah that makes sense - we'll see how this works out in the future :)